benchmarks/pipe.py

```diff
@@ -347,15 +347,19 @@ def __len__(self):
         optimizer.zero_grad()
         try:
             if (pipe_group is None or pipe_group.rank() == 0) and not args.ddp_zero:
-                tmp = batch["input"].to(get_first_device(model))
+                tmp = batch["input"]
+                if torch.cuda.is_available():
+                    tmp = tmp.to(get_first_device(model))
```

Comment:
Can you instead modify get_first_device to return "cpu"?

Reply:
Updated